py-polars/src/series/export.rs

```diff
@@ -23,40 +23,40 @@ impl PySeries {
     /// Non-nullable types are handled with `view()`.
     /// This will cast to floats so that `None = np.nan`.
     fn to_numpy(&self, py: Python) -> PyResult<PyObject> {
+        use DataType::*;
         let s = &self.series;
         match s.dtype() {
-            dt if dt.is_numeric() => {
-                if s.bit_repr_is_large() {
-                    let s = s.cast(&DataType::Float64).unwrap();
-                    let ca = s.f64().unwrap();
-                    let np_arr =
-                        PyArray1::from_iter(py, ca.iter().map(|opt_v| opt_v.unwrap_or(f64::NAN)));
-                    Ok(np_arr.into_py(py))
-                } else {
-                    let s = s.cast(&DataType::Float32).unwrap();
-                    let ca = s.f32().unwrap();
-                    let np_arr =
-                        PyArray1::from_iter(py, ca.iter().map(|opt_v| opt_v.unwrap_or(f32::NAN)));
-                    Ok(np_arr.into_py(py))
-                }
+            Int32 | UInt32 | Int64 | UInt64 | Float64 => {
+                let s = s.cast(&DataType::Float64).unwrap();
+                let ca = s.f64().unwrap();
+                let np_arr =
```

Comment:
There is a double copy for non-float types. I think it's worth it to make a generic and inline cast/ convert to `NaN`. As this will save a double copy for all downstream numpy interop (which I think is a lot).

Reply:
I see what you mean. I can code that :+1: